assets/js/modules/analytics-4/components/common/AdsConversionIDTextField.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Analytics 4 Ads Conversion ID component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { TextField } from 'googlesitekit-components';
+import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
+import VisuallyHidden from '../../../../components/VisuallyHidden';
+import { isValidAdsConversionID } from '../../utils/validation';
+import WarningIcon from '../../../../../svg/icons/warning-v2.svg';
+const { useSelect, useDispatch } = Data;
+
+export default function AdsConversionIDTextField() {
+	const adsConversionID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getAdsConversionID()
+	);
+	const snippetEnabled = useSelect( ( select ) => {
+		return (
+			select( MODULES_ANALYTICS_4 ).getCanUseSnippet() &&
```

Comment:
Just to be clear, we still want to have the `useSnippet` functionality; we just want to remove anything that has to do with `canUseSnippet`.

My only concern is around this temporary fallback mechanism we have around for legacy settings:

https://github.com/google/site-kit-wp/blob/74ca48ba71eb8bd91af1a2d957d8d11c0d4f7696/includes/Modules/Analytics_4/Settings.php#L181-L226

(this is scheduled to be removed as part of #8082 but it got out in 1.118.0 as part of #7923)

At first glance, it doesn't seem so daunting as it just filters the option but doesn't directly change anything in the DB, but suppose, in a scenario where a user had `canUseSnippet` set to `false` in `analytics`; as `getSettings` selector would return the filtered values and `saveSettings` or `submitChanges` actions would save all the settings including the filtered values, there could be an edge case where this legacy setting ends up being migrated into `analytics-4`.

I absolutely agree with you that handing this in #8186 would be the correct course of action by the books, however, do you think my concern above is valid, especially provided that #8186 is being shifted into a new mini-epic and it will likely be deprioritised? As we don't really use `canUseSnippet` anywhere else within `analytics-4`, do you think we can remove these two instances as well (only if you think my concern above is valid)?

1. `assets/js/modules/analytics-4/components/common/AdsConversionIDTextField.js`
2. `assets/js/modules/analytics-4/components/settings/OptionalSettingsView.js`

We do step a little out of our boundaries here in terms of the AC/IB, but I think it might be for a good cause, only if my above concern is valid 😅 

Do let me know what you think, thank you!

Reply:
Just wanted to also add that if we do perform the above change, we could keep #8186 simpler to only remove the infrastructure for `canUseSnippet`, e.g. the module setting in `Analytics_4`, and usage of the legacy `googlesitekit_analytics_can_use_snippet` filter hook in GTM, so the changes would mostly only be in the PHP side.

Reply:
Just to be clear, we still want to have the `useSnippet` functionality; we just want to remove anything that has to do with `canUseSnippet`.

My only concern is around this temporary fallback mechanism we have around for legacy settings:

https://github.com/google/site-kit-wp/blob/74ca48ba71eb8bd91af1a2d957d8d11c0d4f7696/includes/Modules/Analytics_4/Settings.php#L181-L226

(this is scheduled to be removed as part of #8082 but it got out in 1.118.0 as part of #7923)

At first glance, it doesn't seem so daunting as it just filters the option but doesn't directly change anything in the DB, but suppose, in a scenario where a user had `canUseSnippet` set to `false` in `analytics`; as `getSettings` selector would return the filtered values and `saveSettings` or `submitChanges` actions would save all the settings including the filtered values, there could be an edge case where this legacy setting ends up being migrated into `analytics-4`.

I absolutely agree with you that handing this in #8186 would be the correct course of action by the books, however, do you think my concern above is valid, especially provided that #8186 is being shifted into a new mini-epic and it will likely be deprioritised? As we don't really use `canUseSnippet` anywhere else within `analytics-4`, do you think we can remove these two instances as well (only if you think my concern above is valid)?

1. `assets/js/modules/analytics-4/components/common/AdsConversionIDTextField.js`
2. `assets/js/modules/analytics-4/components/settings/OptionalSettingsView.js`

We do step a little out of our boundaries here in terms of the AC/IB, but I think it might be for a good cause, only if my above concern is valid 😅 

If we do perform the above change, we could keep #8186 simpler to only remove the infrastructure for `canUseSnippet`, e.g. the module setting in `Analytics_4`, and usage of the legacy `googlesitekit_analytics_can_use_snippet` filter hook in GTM, so the changes would mostly only be in the PHP side.

Do let me know what you think, thank you!